pkg/jwt/parser.go

```diff
@@ -0,0 +1,43 @@
+package jwt
+
+import (
+	"errors"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+const APPLICATION_TYPE = 2
+
+// TokenClaims represents the role claims presents in the JWT token. This structure extends
+// the jwt.StandardClaims to facilitate the extraction of claims when parsing a JWT token.
+// `Kind` represents the token type in the Mainflux platform and the other properties are
+// part of the RFC specification: https://datatracker.ietf.org/doc/html/rfc7519.
+type TokenClaims struct {
+	Iss  string `json:"iss"`
+	Sub  string `json:"sub"`
+	Kind int    `json:"type"`
+	jwt.StandardClaims
+}
+
+// GetEmail returns the email of the responsible for creating the JWT token, which can be of
+// 'app' or 'user' type. Depending on this, the e-mail can be extract from the `Iss` role claim
+// or the `Sub` role claim.
+func GetEmail(token string) (string, error) {
+	parser := new(jwt.Parser)
+	// second return value can be ignore since it's only the individual parts of the token
+	t, _, err := parser.ParseUnverified(token, &TokenClaims{})
+	if err != nil {
+		return "", errors.New("failed to parse authorization token")
+	}
+
+	claims, ok := t.Claims.(*TokenClaims)
+	if !ok {
+		return "", errors.New("unable to extract token claims")
+	}
```

Comment:
**(non-blocking)** Just to keep it standardized, I'd recommend we keep using `err != nil` :point_down: 

```suggestion
	claims, err := t.Claims.(*TokenClaims)
	if err != nil {
		return "", errors.New("unable to extract token claims")
	}
```

Reply:
The assertion returns a boolean value instead of an error. It's a standard to call it `ok` and handle it in that way. :)

Reply:
Makes sense! :smile: